### PR TITLE
buffers,map_ArrayBuffer:* fake transfer of mapped array buffers

### PR DESCRIPTION
**Pull request by austinEng**
Passes with updated Chromium implementation in https://chromium-review.googlesource.com/c/chromium/src/+/3496685

Issue: #1033

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by kainino0x** [comment]
Issue is #1033 (I just converted the draft to an issue)

File: src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts [hunk 1]
```diff
@@ -1,72 +1,81 @@
...
-      resolveOnTimeout(100),
-      // Either way, if postMessage succeeded, then we'll receive an ArrayBuffer on port2, and this
-      // will win the race.
-      ab2Promise,
-    ]);
+    const ab2 = await ab2Promise;
+    t.expect(ab2.byteLength === kSize, 'ab2 should be the same size');
+    const ab2Data = new Uint32Array(ab2, 0, initialData.length);
+    // ab2 should have the same initial contents.
+    checkElementsEqual(ab2Data, initialData);
```
Comment (kainino0x) [inline src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts:68]: this needs `t.expectOK()` around it (I guess the name isn't clear)

File: src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts [hunk 2]
```diff
@@ -1,72 +1,81 @@
...
-      // If `transfer` is false, this is a deep copy, and it shouldn't be affected by the unmap.
-      t.expect(ab2.byteLength === kSize, 'ab2 should be the same size');
+    // Mutations to ab2 should not be visible in ab1.
+    const ab1Data = new Uint32Array(ab1, 0, initialData.length);
+    const abs2NewData = initialData.slice().reverse();
+    for (let i = 0; i < ab2Data.length; ++i) {
+      ab2Data[i] = abs2NewData[i];
     }
+    checkElementsEqual(ab1Data, initialData);
+    checkElementsEqual(ab2Data, abs2NewData);
```
Comment (kainino0x) [inline src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts:77]: same

**Review by kainino0x — APPROVED** [review]

File: src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts (hunk 1 above)
Comment (austinEng) [inline src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts:68]:
```suggestion
    t.expectOK(checkElementsEqual(ab2Data, initialData));
```

File: src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts (hunk 2 above)
Comment (austinEng) [inline src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts:77]:
```suggestion
    t.expectOK(checkElementsEqual(ab1Data, initialData));
    t.expectOK(checkElementsEqual(ab2Data, abs2NewData));
```

**Pull request merged**